tsdb/index.go

```diff
@@ -1982,7 +1982,8 @@ func (itr *measurementSeriesKeyByExprIterator) Next() ([]byte, error) {
 }
 
 func (itr *measurementSeriesKeyByExprIterator) Close() error {
-	if itr == nil {
```

Comment:
Do you want to get rid of the `itr == nil` check? Would that be impossible? Having both seems more defensive.

Reply:
I think I'd prefer the panic - nobody should be using a nil, and we usually don't check it I think.

Reply:
Dave Cheney is on my side here, which is good enough for me:

https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_never_use_nil_to_indicate_failure

> Tip: Don’t check for a nil receiver, employ high test coverage and vet/lint tools to spot unhandled error conditions resulting in nil receivers.

Reply:
Unfortunately, we have that panic happening, and I have to figure it out. It looks like a failed cleanup from the creation of multiple iterators, which are all then closed, and some are nil while others were created.....